tests/network/probes.go

```diff
@@ -327,10 +327,10 @@ func serverStarter(vmi *v1.VirtualMachineInstance, probe *v1.Probe, port int) {
 	}
 }
 
-func pointProbeToSupportPod(pod *corev1.Pod, IPFamily corev1.IPFamily, probe *v1.Probe) (*v1.Probe, error) {
-	supportPodIP := libnet.GetPodIpByFamily(pod, IPFamily)
+func pointIpv6ProbeToSupportPod(pod *corev1.Pod, probe *v1.Probe) (*v1.Probe, error) {
```

Comment:
I do not understand why this is specific to IPv6. Just because it was called only when the family is IPv6?

Reply:
Well, I'm only halfway into this, and I did not speak with anyone about it, but it seems that the "supportPod" is a kind of mock to allow readiness checking over IPv6 even though something. I may well be wrong, but it smells.

As you say, it is called only in IPv6 context. I'm trying to make everything here thinner and simpler and more specific in order to be able to see what is really needed.

Reply:
I find it odd that we need to make it IPv6 specific. The action itself is generic and taking a step forward to make it specific does not make it clearer to me.
Said that, if you feel it does make it clearer, I have no strong opinions on this.

The support pod, if I recall correctly, is a hack to check IPv6 probes on a dual stack cluster, where IPv4 is the primary family. If no Host is given to a probe, it uses the pod (first) IP. We want to use the second IP, which is IPv6, but have a problem to do so on a VMI, because we do not know the IPv6 address yet. Therefore, we create a helper pod, read its IPv6 address, and use that as the probe Host.
Yes, it is a mess. We could remove it when single stack IPv6 is ready, testing this only there.

Reply:
> The support pod, if I recall correctly, is a hack to check IPv6 probes on a dual stack cluster...

That's basically what I'm saying. It's a hack. A mock. Something that we use in the test because we cannot test the real thing, and it's limited to IPv6. I find the code simpler when it does not pretend to support both IP versions.